observe-api/rust/src/lib.rs

```diff
@@ -28,23 +28,23 @@ pub enum MetricFormat {
 
 pub fn metric(format: MetricFormat, message: &str) {
     let format = format as u32;
-    let ptr = message.as_ptr() as *const u8;
-    let ptr = ptr as u64;
+    let ptr = message.as_ptr();
+    let ptr = ptr as u32;
     let len = message.len() as u32;
     unsafe { _metric(format, ptr, len) };
 }
 
 pub fn span_enter(name: &str) {
-    let ptr = name.as_ptr() as *const u8;
-    let ptr = ptr as u64;
+    let ptr = name.as_ptr();
+    let ptr = ptr as u32;
     let len = name.len() as u32;
     unsafe { _span_enter(ptr, len) };
 }
 
 pub fn span_tags(tags: Vec<&str>) {
     let tags = tags.join(",");
-    let ptr = tags.as_ptr() as *const u8;
-    let ptr = ptr as u64;
+    let ptr = tags.as_ptr();
+    let ptr = ptr as u32;
```

Comment:
We're using the component API representations of lists and strings - two u32 values (instead of a u64 offset and a u32 extent)